### ISSUE TITLE
graphql-XXX-transformer packages depend on graphql-transformer-core and others using exact semvers

### ISSUE THREAD

**Issue by ankon** [aws-amplify/amplify-cli#6268]
**Describe the bug**

I'm trying to depend on multiple graphql-XXX-transformer packages and the graphql-transformer-core package, so I can configure our own transformers in addition to some amplify transformers.

I need to pin down every version and explicitly run `npm dedupe`, because the transformers all use exact-version dependencies, and would get their own copy of the dependency -- but then compilation of my code breaks with errors such as this one:

~~~~
Type 'import(".../node_modules/graphql-transformer-core/lib/TransformerContext").TransformerContextMetadata' is not assignable to type 'import(".../node_modules/graphql-key-transformer/node_modules/graphql-transformer-core/lib/TransformerContext").TransformerContextMetadata'.
~~~~

**Amplify CLI Version**

The project is not using the amplify CLI directly, but has these amplify-cli dependencies:
~~~~
├── amplify-cli-core@1.12.0 
├─┬ graphql-auth-transformer@6.23.1 
│ ├── graphql-connection-transformer@4.19.3  deduped
│ └── graphql-transformer-core@6.25.0  deduped
├─┬ graphql-connection-transformer@4.19.3 
│ ├── graphql-dynamodb-transformer@6.21.3  deduped
│ ├─┬ graphql-key-transformer@2.20.3 
│ │ ├── graphql-dynamodb-transformer@6.21.3  deduped
│ │ └── graphql-transformer-core@6.25.0  deduped
│ └── graphql-transformer-core@6.25.0  deduped
├─┬ graphql-dynamodb-transformer@6.21.3 
│ └── graphql-transformer-core@6.25.0  deduped
├─┬ graphql-key-transformer@2.20.4 
│ ├─┬ graphql-dynamodb-transformer@6.21.4 
│ │ └── graphql-transformer-core@6.25.1  deduped
│ └─┬ graphql-transformer-core@6.25.1 
│   └── amplify-cli-core@1.11.0 
└── graphql-transformer-core@6.25.0 
~~~~

**Expected behavior**

npm should be able to deduplicate the dependencies, even with minor/patch-level version differences.

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu]: Fedora
 - Node Version: v14.15.1

**Additional context**

**Comment by kaustavghosh06** [comment]
This has been fixed as a part of #6442 and would be a part of our next release.

**Comment by ankon** [comment]
Quick check, looks better already I think: Without running a dedupe explicitly, and with adding a '^' in front of all dependencies I now see this
~~~~
$ grep graphql- package.json  | cut -f 1 -d : | xargs npm ls amplify-cli-core
├── amplify-cli-core@1.14.1 
├─┬ graphql-auth-transformer@6.23.3 
│ ├── graphql-connection-transformer@4.19.4  deduped
│ └── graphql-transformer-core@6.25.1  deduped
├─┬ graphql-connection-transformer@4.19.4 
│ ├── graphql-dynamodb-transformer@6.21.4  deduped
│ ├── graphql-key-transformer@2.20.4  deduped
│ └── graphql-transformer-core@6.25.1  deduped
├─┬ graphql-dynamodb-transformer@6.21.4 
│ └── graphql-transformer-core@6.25.1  deduped
├─┬ graphql-function-transformer@2.4.4 
│ └── graphql-transformer-core@6.25.1  deduped
├─┬ graphql-key-transformer@2.20.4 
│ ├── graphql-dynamodb-transformer@6.21.4  deduped
│ └── graphql-transformer-core@6.25.1  deduped
└─┬ graphql-transformer-core@6.25.1 
  └── amplify-cli-core@1.11.0 
~~~~

Still suspicious is the dependency from graphql-transformer-code to amplify-cli-core though.
